### PR TITLE
Implement update_court_outcome_action classification for cases [MA-254]

### PR DESCRIPTION
**Pull request by georgeschena**
### WHAY
Add new classification `update_court_outcome_action `

### WHY
So users can see the latest court actions for a case

### CODE REVIEW

File: spec/lib/hackney/income/tenancy_prioritiser/classifications/apply_for_court_date_spec.rb [hunk 1]
```diff
@@ -109,7 +109,7 @@
       eviction_date: nil
     },
     {
-      outcome: :apply_for_court_date,
+      outcome: :update_court_outcome_action,
```
Comment (kjdchapman) [inline spec/lib/hackney/income/tenancy_prioritiser/classifications/apply_for_court_date_spec.rb]: Should these cases still be part of the `apply_for_court_date_spec`? 

What's the best way to organise these examples?

File: spec/lib/hackney/income/tenancy_prioritiser/classifications/update_court_outcome_spec.rb [hunk 2]
```diff
@@ -0,0 +1,78 @@
+require 'rails_helper'
+
+describe 'Update court outcome', type: :feature do
```
Comment (kjdchapman) [inline spec/lib/hackney/income/tenancy_prioritiser/classifications/update_court_outcome_spec.rb]: What did you think about the style of testing in https://github.com/LBHackney-IT/lbh-income-api/pull/289? 

Is writing examples in this style a better way to express these "update court outcome" examples?

File: app/models/hackney/income/models/case_priority.rb [hunk 3]
```diff
@@ -4,7 +4,8 @@ module Models
       class CasePriority < ApplicationRecord
         enum classification: {
           no_action: 0, send_letter_two: 1, send_letter_one: 2, send_first_SMS: 3, send_NOSP: 4,
-          apply_for_court_date: 6, send_court_warning_letter: 7, send_court_agreement_breach_letter: 8
+          apply_for_court_date: 6, send_court_warning_letter: 7, update_court_outcome_action: 8,
+          send_court_agreement_breach_letter: 8
```
Comment (richrace) [inline app/models/hackney/income/models/case_priority.rb]: Are both `update_court_outcome_action` and `send_court_agreement_breach_letter` meant to be `8`?

File: spec/lib/hackney/income/tenancy_prioritiser/classifications/apply_for_court_date_spec.rb (hunk 1 above)
Comment (georgeschena) [reply spec/lib/hackney/income/tenancy_prioritiser/classifications/apply_for_court_date_spec.rb]: I feel separating each outcome into a separate spec to make it easy to see where the tests are for each outcome imo.

File: spec/lib/hackney/income/tenancy_prioritiser/classifications/update_court_outcome_spec.rb (hunk 2 above)
Comment (georgeschena) [reply spec/lib/hackney/income/tenancy_prioritiser/classifications/update_court_outcome_spec.rb]: I don't really have a strong preference, It didn't make it any easier for me to see the test failing, but I guess it does reduce code duplication which is always nice

Comment (georgeschena) [reply spec/lib/hackney/income/tenancy_prioritiser/classifications/update_court_outcome_spec.rb]: Went with this design choice after speaking to Kyle

File: spec/lib/hackney/income/tenancy_prioritiser/classifications/apply_for_court_date_spec.rb [hunk 4]
```diff
@@ -61,12 +62,12 @@
...
       description: 'with a past court date',
       outcome: :apply_for_court_date,
-      courtdate: 5.days.ago.to_date
+      courtdate: ''
     ),
     base_example.merge(
       description: 'with an upcoming court date',
       outcome: :no_action,
-      courtdate: 2.weeks.from_now.to_date
+      eviction_date: 1.day.from_now.to_date
```
Comment (richrace) [inline spec/lib/hackney/income/tenancy_prioritiser/classifications/apply_for_court_date_spec.rb]: Description isn't correct?

File: spec/lib/hackney/income/tenancy_prioritiser/classifications/update_court_outcome_spec.rb [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    court_outcome: '',
+    courtdate: Date.today - 14.days
+  }
+
+  examples = [
+    base_example,
+    base_example.merge(
+      description: 'with a court date in the past by 100 days',
+      outcome: :update_court_outcome_action,
+      courtdate: Date.today - 100.days
```
Comment (richrace) [inline spec/lib/hackney/income/tenancy_prioritiser/classifications/update_court_outcome_spec.rb]:
```suggestion
      courtdate: 100.days.ago.to_date
```

File: spec/lib/hackney/income/tenancy_prioritiser/classifications/update_court_outcome_spec.rb [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    ),
+    base_example.merge(
+      description: 'with a court date in the past by 100 days',
```
Comment (richrace) [inline spec/lib/hackney/income/tenancy_prioritiser/classifications/update_court_outcome_spec.rb]: Duplicate.

File: spec/lib/hackney/income/tenancy_prioritiser/classifications/update_court_outcome_spec.rb [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+      outcome: :update_court_outcome_action,
+      courtdate: Date.today - 2.days
```
Comment (richrace) [inline spec/lib/hackney/income/tenancy_prioritiser/classifications/update_court_outcome_spec.rb]:
```suggestion
      courtdate: 2.days.ago.to_date
```

File: spec/lib/hackney/income/tenancy_prioritiser/classifications/update_court_outcome_spec.rb [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+    ),
+    base_example.merge(
+      description: 'with a court date 14 days in the future',
+      outcome: :no_action,
+      courtdate: Date.today + 14.days
```
Comment (richrace) [inline spec/lib/hackney/income/tenancy_prioritiser/classifications/update_court_outcome_spec.rb]:
```suggestion
      courtdate: 14.days.from_now.to_date
```

**Pull request merged**